Source/swiftlint/Helpers/LintOrAnalyzeCommand.swift

```diff
@@ -115,6 +121,21 @@ struct LintOrAnalyzeCommand {
         guard numberOfSeriousViolations == 0 else { exit(2) }
     }
 
+    private static func baseline(_ options: LintOrAnalyzeOptions) throws -> Baseline? {
+        if let baselinePath = options.baseline {
+            do {
+                return try Baseline(fromPath: baselinePath)
+            } catch {
+                Issue.baselineNotReadable(path: baselinePath).print()
+                if options.writeBaseline == options.baseline {
+                    return nil
+                }
+                throw error
```

Comment:
Does it work if we throw `Issue.baselineNotReadable(path: baselinePath)` here instead of printing it dedicatedly?

Reply:
So it would work, but by throwing the error we get more detailed error messaging as well.

```
$ swiftlint --baseline BBB.json
warning: Cannot open or read the baseline file at path 'BBB.json'.
Error: The file “BBB.json” couldn’t be opened because there is no such file.
$ touch BBB.json
$ chmod -r  BBB.json
$
$ swiftlint --baseline BBB.json
warning: Cannot open or read the baseline file at path 'BBB.json'.
Error: The file “BBB.json” couldn’t be opened because you don’t have permission to view it.
```

Also, in the special case where the user is writing to and reading from the same baseline, we print the warning, but proceed anyway, as the next time they invoke the command it will work.

Reply:
Okay, got that. But isn't this also happening due to the `throw error` (when `nil` isn't returned)?

Reply:
So everyone will see

`warning: Cannot open or read the baseline file at path 'BBB.json'.`

if the input and output baseline are different filenames, then `throw error` will happen and SwiftLint will quit with the more detailed error message. e.g.

`Error: The file “BBB.json” couldn’t be opened because there is no such file.`

So people who just get the warning do not get the detail. This is probably not going to happen much, but I guess really we should only be ignoring that if the file doesn't exist. If the file is not writable, the user probably does care.

Reply:
I've rejigged this slightly, so we are now explicitly checking for `CocoaError.fileReadNoSuchFile` in the warning only case.